src/test/java/org/isf/supplier/test/Tests.java

```diff
@@ -77,28 +81,100 @@ public void testSupplierSaveOrUpdate() throws Exception {
 	}
 
 	@Test
-	public void testSupplierGetByID() throws Exception {
+	public void testIoSupplierGetByID() throws Exception {
 		int code = _setupTestSupplier(false);
 		Supplier foundSupplier = supplierIoOperation.getByID(code);
 		_checkSupplierIntoDb(foundSupplier.getSupId());
 	}
 
 	@Test
-	public void testSupplierGetAll() throws Exception {
+	public void testIoSupplierGetAll() throws Exception {
 		int code = _setupTestSupplier(false);
 		Supplier foundSupplier = supplierIoOperation.getByID(code);
-		List<Supplier> suppliers = supplierIoOperation.getAll();
+		List<Supplier> suppliers = supplierBrowserManager.getAll();
 		assertThat(suppliers).contains(foundSupplier);
 	}
 
 	@Test
-	public void testSupplierGetList() throws Exception {
+	public void testIoSupplierGetList() throws Exception {
 		int code = _setupTestSupplier(false);
 		Supplier foundSupplier = supplierIoOperation.getByID(code);
-		List<Supplier> suppliers = supplierIoOperation.getList();
+		List<Supplier> suppliers = supplierBrowserManager.getList();
+		assertThat(suppliers).contains(foundSupplier);
+	}
+
+	@Test
+	public void testMgrSupplierSaveOrUpdate() throws Exception {
+		Supplier supplier = testSupplier.setup(true);
+		assertThat(supplierBrowserManager.saveOrUpdate(supplier)).isTrue();
+		_checkSupplierIntoDb(supplier.getSupId());
+	}
+
+	@Test
+	public void testMgrSupplierGetByID() throws Exception {
+		int code = _setupTestSupplier(false);
+		Supplier foundSupplier = supplierBrowserManager.getByID(code);
+		_checkSupplierIntoDb(foundSupplier.getSupId());
+	}
+
+	@Test
+	public void testMgrSupplierGetAll() throws Exception {
+		int code = _setupTestSupplier(false);
+		Supplier foundSupplier = supplierBrowserManager.getByID(code);
+		List<Supplier> suppliers = supplierBrowserManager.getAll();
+		assertThat(suppliers).contains(foundSupplier);
+	}
+
+	@Test
+	public void testMgrSupplierGetList() throws Exception {
+		int code = _setupTestSupplier(false);
+		Supplier foundSupplier = supplierBrowserManager.getByID(code);
+		List<Supplier> suppliers = supplierBrowserManager.getList();
 		assertThat(suppliers).contains(foundSupplier);
 	}
 
+	@Test
+	public void testMgrGetHashMap() throws Exception {
+		int code = _setupTestSupplier(false);
+		Supplier foundSupplier = supplierBrowserManager.getByID(code);
+		// get all (including deleted)
+		HashMap<Integer, String> allSuppliers = supplierBrowserManager.getHashMap(true);
+		// get all (not including deleted)
+		HashMap<Integer, String> suppliers = supplierBrowserManager.getHashMap(true);
+		assertThat(allSuppliers).isEqualTo(suppliers);
+	}
```

Comment:
I am not sure what are you attempting here,  the HashMaps asserted as equal `allSuppliers ` and `suppliers`  are both coming from a call to same method, with same `true` parameter provided, and no logic between calls.

Also `foundSupplier`  is not being used.

Reply:
@emecas Good catch.  I've updated the test after correcting the *getHashMap()* method in the manager class 😄 (see this [JIRA issue](https://openhospital.atlassian.net/browse/OP-407) for more information).